src/Views/PDFPreview/Index.cshtml

```diff
@@ -0,0 +1,116 @@
+<div class="alert alert-info" role="alert">
+  Experimental preview using the local PDF generator.
+</div>
+
+
+<div id="missingParamsError" class="alert alert-error" style="display: none;" role="alert">
+  Query parameters org, app and instance are required.
+</div>
+
+<div id="missingCookieError" class="alert alert-error" style="display: none;" role="alert">
+  Authentication failed, please try the "Refresh authentication" button.
+</div>
+
+<div id="failedError" class="alert alert-error" style="display: none;" role="alert">
+  PDF generation failed, refresh to try again.
+</div>
+
+<iframe id="pdfIframe" title="" style="display: none; width: 100%; height: 100vh; margin-bottom: -16px"></iframe>
+
+<script>
+  function generatePDF() {
+    let params = new URLSearchParams(window.location.search);
+    let org = params.get('org');
+    let app = params.get('app');
+    let instance = params.get('instance');
+    let language = params.get('lang') ?? 'nb';
+
+    if (!org || !app || !instance){
+      document.querySelector('#missingParamsError').style.display = 'block';
+      return;
+    }
+
+    let cookies = document.cookie
+    .split(';')
+    .map(v => v.split('='))
+    .reduce((acc, v) => {
+      acc[decodeURIComponent(v[0].trim())] = decodeURIComponent(v[1].trim());
+      return acc;
+    }, {});
+
+    let token = cookies['AltinnStudioRuntime'];
+    let partyId = cookies['AltinnPartyId'];
```

Comment:
Why do you need AltinnPartyId? Isn’t that only used as a suggestion for what party to new instances should be owned by?

Reply:
You are right, its not used in app-lib so probably not necessary 👍